sources/engine/Stride.BepuPhysics/Stride.BepuPhysics.Navigation/Processors/RecastMeshSystem.cs

```diff
@@ -335,22 +339,27 @@ public static DtStatus FindFollowPath(DtNavMeshQuery navQuery, long startRef, lo
 
         smoothPath.Clear();
         smoothPath.Add(iterPos.ToStrideVector());
-        var visited = new List<long>();
+
+        Span<long> visited = stackalloc long[16];
```

Comment:
Should this length be configurable ? I see that the array is filled in `MoveAlongSurface` and any overflow is discarded in, but not sure if `MergeCorridorStartMoved` would cause issues if it is missing values ?

Reply:
It definitely could be, I'm probably going to have to add it to the navigation settings object for gamesettings? I will actually have to clean that one up as well at some point since I dont think all of the settings are being used properly or at all.